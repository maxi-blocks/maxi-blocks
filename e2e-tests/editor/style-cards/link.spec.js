/**
 * WordPress dependencies
 */
import { createNewPost, setBrowserViewport } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	getStyleCardEditor,
	editGlobalStyles,
	checkSCResult,
	copySCToEdit,
} from '../../utils';

describe('SC Link', () => {
	it('Checking link accordion', async () => {
		await createNewPost();
		await setBrowserViewport('large');

		await getStyleCardEditor({
			page,
			accordion: 'link',
		});
		await copySCToEdit(page, `copy - ${Date.now()}`);

		// Link Colour
		await editGlobalStyles({
			page,
			block: 'link',
			type: 'link',
		});

		// hover Colour
		await editGlobalStyles({
			page,
			block: 'link',
			type: 'hover',
		});

		// active Colour
		await editGlobalStyles({
			page,
			block: 'link',
			type: 'active',
		});

		// visited Colour
		await editGlobalStyles({
			page,
			block: 'link',
			type: 'visited',
		});

		expect(await checkSCResult(page)).toMatchSnapshot();
	});
});
