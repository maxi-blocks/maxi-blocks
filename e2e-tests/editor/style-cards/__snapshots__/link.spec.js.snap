// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SC Link Checking link accordion 1`] = `
{
  "link": {
    "active-color-global": true,
    "hover-color-global": true,
    "link-color": "",
    "link-color-global": true,
    "link-palette-color": 4,
    "link-palette-opacity": 1,
    "link-palette-status": true,
    "visited-color-global": true,
  },
}
`;
